### DISCUSSION TITLE
Create a global prisma client instance that works when using SSR in next JS

### DISCUSSION THREAD

**Discussion by younes-alouani** [prisma/prisma#13951] [Q&A]
Here is a code to create a global prisma client instance :

```
import { PrismaClient } from "@prisma/client";

// PrismaClient is attached to the `global` object in development to prevent
// exhausting your database connection limit.
//
// Learn more: 
// https://pris.ly/d/help/next-js-best-practices

let prisma: PrismaClient | undefined

if (typeof window === "undefined") {
  if (process.env.NODE_ENV === "production") {
    prisma = new PrismaClient();
  } else {
    let globalWithPrisma = global as typeof globalThis & {
      prisma: PrismaClient;
    };
    if (!globalWithPrisma.prisma) {
      globalWithPrisma.prisma = new PrismaClient();
    }
    prisma = globalWithPrisma.prisma;
  }
}

export default prisma 
```
You can learn more on https://pris.ly/d/help/next-js-best-practices

_Originally posted by @younes-alouani in https://github.com/prisma/prisma/issues/5795#issuecomment-1159609317_

**Comment by matthewmueller** [comment]
Hey @younes-alouani is this a problem? Or are you sharing this with someone? I'm confused by this issue 😅  

Going to convert to a discussion for now, because I don't think it's a problem.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
